src/mds/MDLog.cc

```diff
@@ -604,11 +614,33 @@ void MDLog::try_to_commit_open_file_table(uint64_t last_seq)
   }
 }
 
-void MDLog::trim(int m)
+void MDLog::log_trim_upkeep(void) {
+  dout(10) << dendl;
+
+  submit_mutex.lock();
+  while (!upkeep_log_trim_shutdown.load()) {
+    {
+      // adjust lock order
+      submit_mutex.unlock();
+      std::scoped_lock mlock(mds->mds_lock);
+      submit_mutex.lock();
+
+      if (mds->is_active() || mds->is_stopping()) {
+	trim();
+      }
+    }
+
+    auto now = ceph::coarse_mono_clock::now();
```

Comment:
only used once, why not just add it directly to `when`?

Reply:
formatting wise as well it won't be an issue since the big line `ceph::make_timespan(g_conf().get_val<std::chrono::seconds>("mds_log_trim_upkeep_interval").count())` is anyways going to stay on the line below it

Reply:
the compiler would optimise it and its better to read.